bandit/plugins/general_hardcoded_password.py

```diff
@@ -211,5 +216,5 @@ def hardcoded_password_default(context):
     for key, val in zip(context.node.args.args, defs):
         if isinstance(key, ast.Name) or isinstance(key, ast.arg):
             check = key.arg if sys.version_info.major > 2 else key.id  # Py3
-            if isinstance(val, ast.Str) and check in CANDIDATES:
+            if isinstance(val, ast.Str) and RE_CANDIDATES.search(check):
```

Comment:
What if someone does `subprocess.check_output('pwd')`? This would be triggered right? Perhaps we should lower the confidence for this particular check or split them up?

Reply:
Hi @sigmavirus24 the strings "are not passwords hardcoded in any case" for this rules.
Only the kwargs matching with the function definition will raise this detection.

For example:
```
def my_function(password):
    print(password + 'is a pasword')

my_function('hardcode')
```

That code will not be detected as insecure.